odxtools/diaglayer.py

```diff
@@ -256,60 +338,141 @@ def diag_data_dictionary_spec(self) -> DiagDataDictionarySpec:
-        This computes the services that are inherited from other diagnostic layers."""
-        result_dict = {}
+        """
+
+        result_dict: Dict[str, T] = {}
 
-        # Look in parent refs for inherited services Fetch services
-        # from low priority parents first, then update with increasing
-        # priority
+        # populate the result dictionary with the inherited objects
+        #
+        # TODO (?): make sure that there are no "illegal" collisions
+        # i.e., different objects with the same short name stemming
+        # from parent layers exhibiting the same priority that are not
+        # overwritten by a locally defined object. (IMO, this is quite
+        # a corner case.)
         for parent_ref in self._get_parent_refs_sorted_by_priority():
-            for service in parent_ref.get_inherited_services():
-                result_dict[service.short_name] = service
+            parent_dl = parent_ref.layer
+            for dc in parent_dl._compute_available_objects(get_local_objects, get_not_inherited):
+                result_dict[dc.short_name] = dc  # type: ignore[attr-defined]
 
-        for service in self.__gather_local_services(odxlinks):
-            result_dict[service.short_name] = service
+            # remove the explictly not inherited objects
+            for sn in get_not_inherited(parent_ref):
+                if sn in result_dict:
+                    del result_dict[sn]
 
-        return list(result_dict.values())
+        # consider the locally defined objects (override the
+        # inherited entries or add new ones)
+        for obj in get_local_objects(self):
+            result_dict[obj.short_name] = obj  # type: ignore[attr-defined]
 
-    def _compute_available_data_object_properties(self) -> List[DataObjectProperty]:
-        """Returns the locally defined and inherited DOPs."""
-        result_dict = {}
+        return result_dict.values()
 
-        # Look in parent refs for inherited DOPs. Fetch the DOPs from
-        # low priority parents first, then update with increasing
-        # priority
-        for parent_ref in self._get_parent_refs_sorted_by_priority():
-            for dop in parent_ref.get_inherited_data_object_properties():
-                result_dict[dop.short_name] = dop
+    def _get_local_diag_comms(self, odxlinks: OdxLinkDatabase
+                             ) -> Iterable[Union[DiagService, SingleEcuJob]]:
+        """Return the list of locally defined diagnostic communications.
 
-        if self.local_diag_data_dictionary_spec:
-            for dop in self.local_diag_data_dictionary_spec.all_data_object_properties:
-                result_dict[dop.short_name] = dop
+        This is not completely trivial as it requires to resolving the
+        references specified in the <DIAG-COMMS> XML tag.
+        """
+        result_dict: Dict[str, Union[DiagService, SingleEcuJob]] = {}
+
+        # TODO (?): add objects from the import-refs
```

Comment:
Forgot about this?

Reply:
not really, this is more a reminder for future me (or other contributors): I have not come across files that which import-refs, and I am not 100% sure about their semantics...